codice-itest-impl/src/main/java/org/codice/itest/reporter/RemainingTestsReporter.java

```diff
@@ -0,0 +1,23 @@
+package org.codice.itest.reporter;
+
+import org.codice.itest.api.TestResult;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+public class RemainingTestsReporter implements Consumer<TestResult> {
+    private List<String> testNames;
+    public RemainingTestsReporter(List<String> testNames){
+        this.testNames = testNames;
+    }
+
+    @Override
+    public void accept(TestResult testResult) {
+        if(testNames.contains(testResult.getTestName())){
+            testNames.remove(testResult.getTestName());
```

Comment:
:exclamation: this isn't threadsafe. You're modifying a list that's shared with other classes. I'd recommend making a copy of the list in the constructor.